src/neuroglancer/datasource/graphene/frontend.ts

```diff
@@ -291,11 +303,53 @@ function parseTransform(data: any): mat4 {
   });
 }
 
+function parseShardingEncoding(y: any): DataEncoding {
```

Comment:
I think in the long term these two should be together because they both do parsing sharding parameters
I did end up changing a lot while experimeting with stuff but keep in mind that this won't be merged with master right away, first with `akhilesh-graphene-sharded` then master